### DISCUSSION TITLE
Excessive network usage and caching inefficiencies in app

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#30093] [Ideas]
It occupies a tiny network every moment. When in the background, there will be continuous 1.4KB/s downloads and 0.85KB/s uploads. When browsing maps in the foreground, it will use a downstream bandwidth of 32-529KB/s (averaging about 200KB/s).
In the past 24 hours, it has used a total of 1.4GB of network data (1.33GB downstream and 69.69MB upstream).

I'm not sure about the reason behind it in the background, but when I randomly selected a difficulty, I noticed that the player's avatar needs to be reloaded every time, and the local score is the same. Should it add a cache or something like that? It's almost impossible for me to see everyone's profile picture immediately. Instead, I have to wait for the circle to turn, then wait for the empty profile picture box, and finally it will appear. This happens every time I switch difficulty, and they require broadband consumption.

Every networking event requires waiting, including starting the game, viewing the score avatar, opening one's personal page, Direct Messages, etc, They consume more network and time in certain areas or when the network is poor.

Wait, I noticed something with Ctrl+F3. It looks like everything is already in memory, but why do we have to reload it from the network

**Comment by peppy** [comment]
We'll probably add more caching eventually, sure.

But the background traffic is pretty small so I wouldn't call that an issue.
